### PR TITLE
PIN-3977: removed visibility checks on clients and keys

### PR DESCRIPTION
**Pull request by AsterITA**

### CODE REVIEW

File: src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala [hunk 1]
```diff
@@ -127,26 +127,10 @@ final case class ClientApiServiceImpl(
...
-      clientUuid           <- clientId.toFutureUUID
-      client               <- authorizationManagementService.getClient(clientUuid)
-      isConsumerOrProducer <- isConsumerOrProducer(requesterUuid, client)
-      _                    <- Future
-        .failed(OrganizationNotAllowedOnClient(client.id.toString, requesterUuid))
-        .unlessA(isConsumerOrProducer)
-
+      requesterUuid <- getOrganizationIdFutureUUID(contexts)
+      clientUuid    <- clientId.toFutureUUID
+      client        <- authorizationManagementService.getClient(clientUuid)
```
Comment (galales) [inline src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala:133]: as per task description, relationships should be hidden if the requester is not the consumer

**Review by galales — CHANGES_REQUESTED** [review]

File: src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala [hunk 2]
```diff
@@ -127,26 +127,10 @@ final case class ClientApiServiceImpl(
...
     } yield client.toApi(requesterUuid == client.consumerId)
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala:134]: The implementation is missing the part that manages `relationshipIds` inside the client.
The requirement is:
- if `organizationId` is **not the consumer** then `relationshipIds` field must be set to `List.empty`
- else retrieve the client unchanged

File: src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala [hunk 3]
```diff
@@ -308,8 +290,6 @@ final case class ClientApiServiceImpl(
     val result: Future[Keys] = for {
       clientUuid    <- clientId.toFutureUUID
       relationships <- parseArrayParameters(relationshipIds).traverse(_.toFutureUUID)
-      client        <- authorizationManagementService.getClient(clientUuid)
-      _             <- assertIsClientConsumer(client).toFuture
```
Comment (beetlecrunch) [inline src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala]: This control must be kept.

File: src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala (hunk 2 above)
Comment (AsterITA) [reply src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala:134]: > The implementation is missing the part that manages `relationshipIds` inside the client. The requirement is:
> 
>     * if `organizationId` is **not the consumer** then `relationshipIds` field must be set to `List.empty`
> 
>     * else retrieve the client unchanged

Isn't that part implemented in `client.toApi(requesterUuid == client.consumerId)`? @galales 

Where, inside the `toApi` it sets the relationshiIds based on the resulted boolean: 
`relationshipsIds = if (showRelationShips) p.relationships else Set.empty,`

Comment (galales) [reply src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala:134]: you're right, I forgot it was already implemented 😬

Comment (beetlecrunch) [reply src/main/scala/it/pagopa/interop/authorizationprocess/api/impl/ClientApiServiceImpl.scala:134]: Same, sorry!

File: src/test/scala/it/pagopa/interop/authorizationprocess/ClientOperationSpec.scala [hunk 4]
```diff
@@ -145,12 +145,6 @@ class ClientOperationSpec extends AnyWordSpecLike with MockFactory with SpecUtil
         .once()
         .returns(Future.successful(persistentClient.copy(consumerId = anotherConsumerId)))
```
Comment (galales) [inline src/test/scala/it/pagopa/interop/authorizationprocess/ClientOperationSpec.scala:147]: may I ask you to update this test setting a generic id [here](https://github.com/pagopa/interop-be-authorization-process/pull/226/files#diff-b1fc745ead68a7d0cf5bdedf8ec2775c988a92e6716475ab5869cefca06c3f4dL139)?
So we properly test that anyone can access the client, without seeing the relationships

**Review by beetlecrunch — APPROVED** [review]

**Comment by AsterITA** [comment]
Replaced with hotfix in https://github.com/pagopa/interop-be-authorization-process/pull/227

**Pull request closed without merging**